### PR TITLE
[Mamba] Fix accuracy degradation under high concurrency with --enable-mixed-chunk and --mamba-scheduler-strategy extra_buffer

### PR DESCRIPTION
**Pull request by cyyc0310**
## Motivation

This PR fixes a critical accuracy regression in Mamba models mentioned in #22867  and #23075,co-authored by @flyerming 
When running with both` --enable-mixed-chunk` and `--mamba-scheduler-strategy extra_buffer` under high concurrency scenarios ,multiple requests are merged into a single batch, the Mamba-specific tracking state (mamba_track_indices, mamba_mask, mamba_seqlens) was incorrectly reset to None instead of being properly concatenated. This caused significant accuracy degradation:

- GSM8K: Dropped from 93.8% → 87.6% at concurrency=16
- MMLU: Dropped from ~89.5% → 73.3% at concurrency=16
- AIME25: Dropped from ~88.3% → 80.8% at concurrency=16

## Modifications

Add ` _merge_mamba_tracking() ` function to properly concatenates Mamba tracking tensors (mamba_track_indices, mamba_mask, mamba_seqlens) along the batch dimension during batch merge operations

## Accuracy Tests
Tested with Qwen3.5-35B-A3B, tp=2 on H800 GPUs, fa3 attention backend.

Configurations:
- normal: Baseline (no mixed chunk or extra_buffer)
- mix: `--enable-mixed-chunk`
- mamba_extra_buffer: `--mamba-scheduler-strategy extra_buffer`
- mix+mamba_extra_buffer: `--enable-mixed-chunk` `--mamba-scheduler-strategy extra_buffer`

GSM8K Accuracy (concurrency=1):
| Config                 | Before | After |
|------------------------|--------|-------|
| mix                    | 0.927  | 0.938 |
| mamba_extra_buffer     | 0.938  | 0.933 |
| mix+mamba_extra_buffer | 0.938  | 0.938 |

GSM8K Accuracy (concurrency=16):
| Config                 | Before | After |
|------------------------|--------|-------|
| mix                    | 0.940  | 0.941 |
| mamba_extra_buffer     | 0.935  | 0.939 |
| **mix+mamba_extra_buffer** | **0.886**  | **0.937** |

MMLU Accuracy:
| Config                 | Before | After |
|------------------------|--------|-------|
| normal                 | 0.8958 | 0.8945|
| mix                    | 0.8948 | 0.8943|
| mamba_extra_buffer     | 0.8945 | 0.8945|
| **mix+mamba_extra_buffer** | **0.7326** | **0.8947**|

AIME25 Accuracy:
| Config                 | Before | After |
|------------------------|--------|-------|
| normal                 | 0.8833 | 0.8833|
| mix                    | 0.8833 | 0.8917|
| mamba_extra_buffer     | 0.8750 | 0.8750|
| **mix+mamba_extra_buffer** | **0.8083** | **0.8917**|

**Key Improvements for mix+mamba_extra_buffer config:**

- GSM8K (concurrency=16): **+5.1%** accuracy recovery
- MMLU: **+16.2%** accuracy recovery
- AIME25: **+8.3%** accuracy recovery
- All other configurations remain within statistical noise margin , confirming no regression introduced.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Comment by cyyc0310** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

File: python/sglang/srt/managers/schedule_batch.py [hunk 1]
```diff
@@ -2063,6 +2063,47 @@ def _force_track_h(i: int) -> int:
...
+
+        # Decode requests need placeholder entries (no tracking needed).
+        if self.mamba_track_mask is not None:
+            self.mamba_track_mask = torch.cat(
+                [
+                    self.mamba_track_mask,
+                    torch.zeros(other_bs, dtype=torch.bool, device=device),
+                ]
+            )
+        if self.mamba_track_seqlens is not None:
```
Comment (yizhang2077) [inline python/sglang/srt/managers/schedule_batch.py:2129]: do we need do padding in the left part if `self.mamba_track_seqlens` is None? the same question about `self.mamba_track_mask` / `self.mamba_track_indices`

Comment (cyyc0310) [reply python/sglang/srt/managers/schedule_batch.py:2129]: I think there's no need. The `mamba_track_mask/indice/seqlens `of self (prefill) are always properly initialized in prepare_for_extend(). The previous precision regression was caused by the decode side's `mamba_track_indices/mamba_track_mask/mamba_track_seqlens` not being correctly set during the mix — they were simply left as None. The current code fixes this by applying right-side padding for other (decode). Since mix-related functions are only invoked when extend data already exists, and the only operation that clears the three fields to None — filter_batch() — is always applied to the other side, self always holds valid tracking data upon entering the merge. There is no scenario that requires left-side padding.

Comment (cyyc0310) [reply python/sglang/srt/managers/schedule_batch.py:2129]: I added left-padding just now

**Comment by cyyc0310** [comment]
@yizhang2077  Left padding has been added. Could you please review?
